scripts/zkai_attention_derived_d128_block_statement_chain_gate.py

```
@@ -0,0 +1,881 @@
+    return text
+
+
+def _sha(value: Any, label: str) -> str:
+    return _digest(value, label, prefix="sha256")
+
+
+def read_source_bytes(path: pathlib.Path) -> bytes:
+    root = ROOT.resolve()
+    candidate = path if path.is_absolute() else ROOT / path
+    if candidate.is_symlink():
+        raise AttentionDerivedD128BlockStatementChainError(f"source path must not be symlink: {path}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)